pkg/services/ngalert/api/util.go

```diff
@@ -47,7 +48,7 @@ func backendTypeByUID(ctx *models.ReqContext, cache datasources.CacheService) (a
 			return 0, fmt.Errorf("unexpected backend type (%v)", ds.Type)
 		}
 	}
-	return 0, fmt.Errorf("unexpected backend type (%v)", datasourceUID)
+	return 0, errors.New("cannot retrieve data source by its UID")
```

Comment:
```suggestion
	return 0, errors.New("no datasource was found matching the given UID")
```
Nit - IMO the original wording sounds like "backend had a problem looking up datasources" where really we are wanting to tell the user that they entered in a bad value.